### ISSUE TITLE
Metadata display shows one parameter only

### ISSUE THREAD

**Issue by unnamed777** [musselwhizzle/Focus-Points#181]
It seems that the display shows only first item. When I use filter, behavior is the same: result is changing, but only one item is visible.
<details>
  <summary>screenshot</summary>
<img width="738" alt="Screenshot 2024-07-04 at 21 48 59" src="https://github.com/musselwhizzle/Focus-Points/assets/1111833/acb8aa91-baf7-4090-b0d0-fce366816753">
</details>

Lightroom: 13.3.1, macos
Plugin: 02910d8

**Comment by unnamed777** [comment]
Olympus OM-1, Canon R5. Forgot to say, "Open as text" shows all metadata. The issue happens only in plugin window.
[NA309100.dng.zip](https://github.com/user-attachments/files/16104167/NA309100.dng.zip)

**Comment by capricorn8** [comment]
I have imported your image file and I can see all metadara in the plugin window.

Are you running Windows or Mac?

In Lr Plugin Manager you can require the plugin to log debug information ("Logging" section). On a Windows machine, the logfile can be found under C:\Users\username\Documents\LrClassicLogs\libraryLogger.log.

Can you provide your logfile?

**Comment by unnamed777** [comment]
Macos 14.4.1
[libraryLogger.log](https://github.com/user-attachments/files/16106767/libraryLogger.log)

**Comment by capricorn8** [comment]
The logfile looks as it should. Does this happen for all of your images?

Since I don't have access to a Mac computer, I will need to prepare a special version with additional log entries to get an idea what is wrong.

**Comment by capricorn8** [comment]
The logfile looks as it should. Does this happen for all of your images?

Since I don't have access to a Mac computer, I will need to prepare a special version with additional log entries to get an idea what is wrong.

**Comment by capricorn8** [comment]
Sorry for the long delay, I was busy with other stuff.

I have prepared an instrumented version of the script in which I suspect the problem to happen. 
Can you please copy the attached file over the existing one on your computer, delete any existing libraryLogger logfile and run the plugin?

[ShowMetadata.zip](https://github.com/user-attachments/files/16480923/ShowMetadata.zip)

**Comment by unnamed777** [comment]
Sorry for delay, I've read the notification and forget about it. Here is log.
[libraryLogger.log](https://github.com/user-attachments/files/16791765/libraryLogger.log)

**Comment by capricorn8** [comment]
I was able to locate the problem and (hopefully) fix it. Could you please unzip and copy the attached file over the one in your installation of the plugin and verify?

[MetaDataDialog.zip](https://github.com/user-attachments/files/16907941/MetaDataDialog.zip)

**Comment by unnamed777** [comment]
Yes, now it works. But scrolling is utterly slow. It looks like LR is trying to calculate something to render new area (cpu usage jumps up to 105%). If I scroll up and down (strictly vertically) a few times, this area scrolls as it should (like it's cached). I've tried to record a video https://drive.google.com/file/d/1F1srF65-YQTu3HTOmRu5bw-ZPSTkqKEv/view?usp=sharing . Before each swipe I move cursor, otherwise it's hard to see when swiping (scrolling) is started.
I've disabled logs and restored original ShowMetadata.lua, the behavior is the same.

**Comment by capricorn8** [comment]
Thank you! I will proceed to check-in the fix.

The performance issue known and obviously this exists from the first version of the plugin. It only affects MAC, on WIN the performance is blazingly fast. This is yet another case of LR SDK behaving differently on the two platforms (like the issue you previously reported).

I have some input on how this may be solved, I'm currently sitting over this. Hopefully, I will soon have a fix for this as well!

**Comment by capricorn8** [comment]
Fixed in V2.0.0
